hooks/useGraphData.ts

```diff
@@ -56,24 +52,47 @@ const fetchJarsOnGraph = async (
     ) {
       throw new Error("Invalid response structure");
     }
-    return jsonResponse.data.cookieJars;
+    return jsonResponse.data.cookieJars.map((cookieJar: CookieJar) => ({
+      ...cookieJar,
+      chainId,
+      chainName: chainMetadata[chainId].name,
+    }));
   } catch (error) {
     throw new Error(error as unknown as string);
   }
 };
 
-interface UseGraphDataProps {
-  chainId: number;
+const fetchMultipleJars = async ({
+  chainId,
+  sorting,
+}: UseGraphDataProps): Promise<CookieJar[]> => {
+  const queries = Object.keys(endpoints)
+    .filter((e) => (chainId ? Number(e) === chainId : true))
+    .map((e) => fetchJarsOnGraph(Number(e), sorting));
+
+  const dataFromGraphs = await Promise.allSettled(queries);
+
+  const validJars = dataFromGraphs.reduce((acc, result) => {
+    if (result.status === "fulfilled" && Array.isArray(result.value)) {
+      return [...acc, ...result.value];
+    }
+    return acc;
+  }, [] as CookieJar[]);
```

Comment:
Avoid type casting.
If you update to useQueries (linked in another comment), you should better understand the response data for a single query from the graph schema. define the response data and ideally typescript should understand that.

Above and beyond would be an implementation of https://gql-tada.0no.co/ but maybe that should be its own ticket